vulture_os/toolkit/api_parser/trendmicro_visionone/trendmicro_visionone.py

```diff
@@ -56,75 +59,120 @@ def __init__(self, data):
 
-    def __execute_query(self, url_path, query, timeout=10):
+    def __execute_query(self, kind, since, to, timeout=30):
+
         items = []
-        status = True
+        status = False
 
         headers = {'Authorization': 'Bearer ' + self.trendmicro_visionone_token, 'TMV1-Filter': ""}
+        max_retries = 3
+        retry = max_retries
+
+        query = {}
+        url_path = ""
+        link = ""
+        has_next = False
+
+        while not status and retry > 0 and not self.evt_stop.is_set():
+            self.update_lock()
+            try:
+                if not has_next:
+                    # We have to do this only if we're not using the next link
+                    if kind == "alerts":
+                        query, url_path = self._prepare_request_alerts(since, to)
+                    elif kind == "audit":
+                        query, url_path = self._prepare_request_auditlogs(since, to)
+                    elif kind == "oat":
+                        query, url_path = self._prepare_request_OAT(since, to)
+                        headers.update({'TMV1-Filter': "not (riskLevel eq 'info')"})
+
+                    link = self.URL_BASE + url_path
 
-        link = self.URL_BASE + url_path
-        try:
-            while True:
                 logger.info(f"[{__parser__}]:__execute_query: URL: {link} , params: {query}", extra={'frontend': str(self.frontend)})
                 r = requests.get(link, params=query, headers=headers,proxies=self.proxies, verify=self.api_parser_verify_ssl, timeout=timeout)
                 if r.status_code != 200:
                     status = False
-                    logger.error(f"Error on URL: {link} Status: {r.status_code} Reason/Content: {r.content}")
+                    logger.warning(f"Error on URL: {link} Status: {r.status_code} Reason/Content: {r.content}")
                     raise TrendmicroVisionOneAPIError(f"Error on URL: {link} Status: {r.status_code} Reason/Content: {r.content}")
 
-                items.extend(r.json()['items'])
+                request_json = r.json()
+                if kind != "audit" and request_json.get('totalCount') >= MAX_TOTAL_COUNT:
+                    # Note : This attribute is not present in audit logs response
+                    raise requests.exceptions.ReadTimeout(f"Total count = {request_json.get('totalCount')} > {MAX_TOTAL_COUNT} (given max count)")
 
-                if link := r.json().get('nextLink'):
+                items.extend(request_json['items'])
+
+                if link := request_json.get('nextLink'):
                     query = None
-                    self.update_lock()
-                    continue
+                    has_next = True
                 else:
+                    has_next = False
+                    status = True
+            except requests.exceptions.ReadTimeout as e:
+                logger.exception(f"[{__parser__}]:__execute_query: {e}", extra={'frontend': str(self.frontend)})
```

Comment:
seems like this behaviour is not a hard error: prefer using a logger.warning() in that case

```suggestion
                logger.warning(f"[{__parser__}]:__execute_query: {e}", extra={'frontend': str(self.frontend)})
```

Reply:
Done in https://github.com/VultureProject/vulture-gui/pull/530/commits/4eaa28dee3746a0432fe0b13589344c646ba244e